go-client/client/client.go

```diff
@@ -220,25 +182,77 @@ func (client *Client) withToken(ctx context.Context) (context.Context, error) {
 }
 
 // RunScript executes a script on the deephaven server.
-// The script language depends on the scriptLanguage argument passed when creating the client.
+//
+// The script language depends on the argument passed to WithConsole when creating the client.
+// If WithConsole was not provided when creating the client, this will return ErrNoConsole.
 func (client *Client) RunScript(ctx context.Context, script string) error {
+	if client.consoleStub.consoleId == nil {
+		return ErrNoConsole
+	}
+
 	ctx, err := client.consoleStub.client.withToken(ctx)
 	if err != nil {
 		return err
 	}
 
-	f := func() (*apppb2.FieldsChangeUpdate, error) {
-		// TODO: This might fall victim to the double-responses problem if a FetchTablesRepeating
-		// request is already active.
-
-		req := consolepb2.ExecuteCommandRequest{ConsoleId: client.consoleStub.consoleId, Code: script}
-		rst, err := client.consoleStub.stub.ExecuteCommand(ctx, &req)
-		if err != nil {
-			return nil, err
-		} else {
-			return rst.Changes, nil
-		}
+	req := consolepb2.ExecuteCommandRequest{ConsoleId: client.consoleStub.consoleId, Code: script}
+	_, err = client.consoleStub.stub.ExecuteCommand(ctx, &req)
+	if err != nil {
+		return err
 	}
 
-	return client.fieldMan.ExecAndUpdate(f)
+	return nil
+}
+
+// clientOptions holds a set of configurable options to use when creating a client with NewClient.
+type clientOptions struct {
+	scriptLanguage    string // The language to use for server-side scripts. Empty string means no scripts can be run.
+	allowLeakedTables bool   // When true, disables TableHandle finalizers.
+}
+
+func newClientOptions(opts ...ClientOption) clientOptions {
+	options := clientOptions{}
+	for _, opt := range opts {
+		opt.apply(&options)
+	}
+	return options
+}
+
+// A ClientOption configures some aspect of a client connection when passed to NewClient.
+// See the WithXYZ methods for possible client options.
+type ClientOption interface {
+	// apply sets the relevant option in the clientOptions struct.
+	apply(opts *clientOptions)
+}
+
+// A funcDialOption wraps a function that will apply a client option.
+// Inspiration from the grpc-go package.
+type funcDialOption struct {
+	f func(opts *clientOptions)
+}
+
+func (opt funcDialOption) apply(opts *clientOptions) {
+	opt.f(opts)
+}
+
+// WithConsole allows the client to run scripts on the server using the RunScript method and bind tables to variables using BindToVariable.
```

Comment:
So the lang needs to get set to bind variables?  If so, make sure that all of the docs reflect this.